core/src/main/java/org/elasticsearch/Version.java

```diff
@@ -584,6 +591,15 @@ public Version minimumCompatibilityVersion() {
     }
 
     /**
+     * Returns the maximum hypothetical compatible version based on the current
+     * version. Beta and RC releases are always forward compatible with the maximum
+     * hypothetical version.
+     */
+    public Version maximumCompatibilityVersion() {
```

Comment:
that question did cross my mind too, but historically this is what always has happened, so that was the main drive to add it... but yes since the minimum version is checked on both sides, checking the maximum version isn't necessary.